platform/viewer/src/lib/DownloadViewportEngine.js

```diff
@@ -0,0 +1,137 @@
+const DOWNLOAD_ELEMENT_ID = 'download-element';
+const PREVIEW_ELEMENT_ID = 'preview-element';
+
+class DownloadViewportEngine {
+  constructor() {
+    this.$previewElement = null;
+    this.$downloadElement = document.createElement('div');
+    this.$activeViewport = null;
```

Comment:
You should not need DOM references for anything in this component

Reply:
I needed it to know wich is the active ViewPort dom elem and where, inside the React Modal, to put the blob.